devbin/rotate_keys.py

```diff
@@ -87,15 +109,73 @@ def add_new_key(self, k: IAMKey):
         self.keys.insert(0, k)
 
     def list_keys(self) -> str:
-        msg = f'GSA Keys for {self.email}\n'
+        def color_id(k: IAMKey) -> str:
+            if not k.user_managed:
+                color = GREY
+            elif k.expired():
+                color = ERROR_RED
+            elif not k.recently_created():
+                color = WARNING_YELLOW
+            else:
+                color = OK_GREEN
+            return f'{color}{k.id}{ENDC}'
+
+        msg = f'{self}\n'
         for k in self.keys:
-            msg += f'{k.id} \t Created: {k.created_readable} \t Expires: {k.expiration_readable}'
-            matching_secrets = [str(s) for s in self.kube_secrets if s.private_key_id() == k.id]
+            msg += f'{color_id(k)} \t Created: {k.created_readable} \t Expires: {k.expiration_readable}'
+            matching_secrets = [str(s) for s in self.kube_secrets if s.matches_iam_key(k)]
             if len(matching_secrets) > 0:
                 msg += f'\t <== {" ".join(matching_secrets)}'
             msg += '\n'
         return msg
 
+    def rotation_state(self) -> RotationState:
+        active_key = self.active_user_key()
+
+        # Solely system-managed accounts are always up to date
+        if not active_key:
+            return 'up-to-date'
+
+        old_user_keys = self.redundant_user_keys()
+        if active_key.expired():
+            return 'expired'
+        elif active_key.recently_created() and len(old_user_keys) > 0:
+            return 'in-progress'
+        elif not active_key.recently_created() and len(old_user_keys) > 0:
+            return 'ready-for-delete'
+        else:
+            assert len(old_user_keys) == 0
+            return 'up-to-date'
+
+    def rotation_status(self) -> str:
```

Comment:
I'm not a fan of the words status and state meaning almost the same thing. I'd probably have called this `rotation_state_str`

Reply:
Alternatively, RotationState could be an enum with a custom str.